contracts/ColonyTask.sol

```diff
@@ -137,34 +137,50 @@ contract ColonyTask is ColonyStorage, DSMath {
     return taskCount;
   }
 
-  function proposeTaskChange(bytes _data, uint256 _value, uint8 _role) public returns (uint256 transactionId) {
-    var (sig, taskId) = deconstructCall(_data);
-
-    Task storage task = tasks[taskId];
-    require(task.roles[_role].user == msg.sender);
-    require(!task.finalized);
-
-    uint8[2] storage _reviewers = reviewers[sig];
-    require(_reviewers[0] != 0 || _reviewers[1] != 0);
-    require(_reviewers[0] == _role || _reviewers[1] == _role);
-
-    transactionId = IColony(this).submitTransaction(_data, _value, _role);
+  function getTaskChangeNonce() public view returns (uint256) {
+    return taskChangeNonce;
   }
 
-  function approveTaskChange(uint256 _transactionId, uint8 _role) public {
-    Transaction storage _transaction = transactions[_transactionId];
-    bytes memory _data = _transaction.data;
-    var (sig, taskId) = deconstructCall(_data);
+  function executeTaskChange(
+    uint8[] _sigV,
+    bytes32[] _sigR,
+    bytes32[] _sigS,
+    uint256 _value,
```

Comment:
Surely value is always going to be zero in our case? I can't think of a reason it matters, given that we always call against `address(this)`, other than if they signed a transaction that moved more ETH than the colony had (which would cause the call to fail). I'm all for being extensible in the future without changing the interface, but for now, surely it's better to be more explicit about this with e.g. `require(_value==0)`?

Reply:
This was for future extensibility yes but we can restrict tx value to 0 for now